src/Octoshift/Services/BbsApi.cs

```diff
@@ -73,11 +75,63 @@ public virtual async Task<IEnumerable<(int Id, string Key, string Name)>> GetPro
             .ToListAsync();
     }
 
+    public virtual async Task<(int Id, string Key, string Name)> GetProject(string projectKey)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}";
+        var response = await _client.GetAsync(url);
+
+        var project = JObject.Parse(response);
+        return ((int)project["id"], (string)project["key"], (string)project["name"]);
+    }
+
     public virtual async Task<IEnumerable<(int Id, string Slug, string Name)>> GetRepos(string projectKey)
     {
         var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos";
         return await _client.GetAllAsync(url)
             .Select(x => ((int)x["id"], (string)x["slug"], (string)x["name"]))
             .ToListAsync();
     }
+
+    public virtual async Task<bool> GetIsRepositoryArchived(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}?fields=archived";
+        var response = await _client.GetAsync(url);
+
+        var data = JObject.Parse(response);
+        return (bool)data["archived"];
+    }
+
+    public virtual async Task<IEnumerable<(int Id, string Name)>> GetRepositoryPullRequests(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}/pull-requests";
+        return await _client.GetAllAsync(url)
+            .Select(x => ((int)x["id"], (string)x["name"]))
+            .ToListAsync();
+    }
+
+    public virtual async Task<JObject> GetRepositoryLatestCommit(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}/commits?limit=1";
+
+        try
+        {
+            var response = await _client.GetAsync(url);
+
+            var data = JObject.Parse(response);
+            return data;
+        }
+        catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+        {
+            return new JObject();
+        }
+    }
+
+    public virtual async Task<JObject> GetRepositorySize(string projectKey, string repo, string bbsUsername, string bbsPassword)
```

Comment:
Same comment about not returning JObject.

Reply:
Updated in [b83281c](https://github.com/github/gh-gei/pull/1015/commits/b83281c162de64294614f7202857ea1b354f630a)